drivers/ethernet/eth_sam_gmac.c

```diff
@@ -1970,8 +1970,8 @@ static void eth0_iface_init(struct net_if *iface)
 	}
 
 	/* Initialise monitor */
-	k_delayed_work_init(&dev_data->monitor_work, monitor_work_handler);
-	k_delayed_work_submit(&dev_data->monitor_work,
+	k_work_init_delayable(&dev_data->monitor_work, monitor_work_handler);
+	k_work_reschedule(&dev_data->monitor_work,
 			      K_MSEC(CONFIG_ETH_SAM_GMAC_MONITOR_PERIOD));
```

Comment:
nit: there are quite a few places in this PR with broken alignment after the API change, like here, probably worth pushing the changes through some formatter.

Reply:
> there are quite a few places in this PR with broken alignment after the API change, like here, probably worth pushing the changes through some formatter.

Yeah, there are now these style issues in the code. I went through the changes in the `subsys/net/*` and manually changed those. I did not check all the changes in this PR (especially in the drivers/) as there was just so much of it. Using formatter could work except that it can change lines that we not changing in this k_work conversion, so I did not use a formatter. I think this work can be done in a separate PR if really needed.

Reply:
> Using formatter could work except that it can change lines that we not changing in this k_work conversion, so I did not use a formatter.

I typically use `git-clang-format HEAD~1` to apply the formatting only to the changes from the most recent commit, this combines well with an interactive rebase. But I now noticed that while it fixes the alignment issues I spotted it also introduce some others (like for instance changing the commit alignment from tabs to spaces which in effect doesn't look good with other lines aligned with tabs). As it's really not that important I guess it could indeed be fixed at some point in the future in a more comprehensive approach.

Reply:
> use `git-clang-format HEAD~1` to apply the formatting only to the changes from the most recent commit

Thanks for the hint, I was not aware of this tool.

Reply:
> As it's really not that important I guess it could indeed be fixed at some point in the future in a more comprehensive approach.

I will revisit this PR and update the code. I did not realize there was a tool for doing the work and it looked a lot of work to go through the patches manually.